app/lib/service/youtube/backend.dart

```diff
@@ -124,14 +128,29 @@ class _PkgOfWeekVideoFetcher {
           pageToken: nextPageToken,
         );
         videos.addAll(rs.items!.map(
-          (i) => PkgOfWeekVideo(
-            videoId: i.contentDetails!.videoId!,
-            title: i.snippet!.title!,
-            description:
-                (i.snippet?.description ?? '').trim().split('\n').first,
-            thumbnailUrl: i.snippet!.thumbnails!.high!.url!,
-          ),
-        ));
+          (i) {
+            try {
+              final thumbnails = i.snippet!.thumbnails!;
+              final thumbnail = thumbnails.high ??
+                  thumbnails.default_ ??
+                  thumbnails.maxres ??
+                  thumbnails.standard ??
+                  thumbnails.medium;
+              return PkgOfWeekVideo(
+                videoId: i.contentDetails!.videoId!,
+                title: i.snippet?.title ?? '',
+                description:
+                    (i.snippet?.description ?? '').trim().split('\n').first,
+                thumbnailUrl: thumbnail!.url!,
+              );
+            } catch (e, st) {
+              // this item will be skipped, the rest of the list may be displayed
+              _logger.pubNoticeWarning(
```

Comment:
Let's make it shout.. this shouldn't happen... If it does I suggest we filter out entries without a URL..